client/lib/components/recent_numbers_graph.dart

```diff
@@ -75,7 +75,13 @@ class _RecentNumbersBarGraphState extends State<RecentNumbersBarGraph> {
               final date = DateTime.fromMillisecondsSinceEpoch(
                   widget.timeseries[index].epochMsec.toInt());
               // Abbr to fit on single line: e.g. "Oct 18, 2020"
-              final formattedDate = DateFormat.yMMMd().format(date);
+              final formattedDate = DateFormat.yMMMd().format(
```

Comment:
Extract this to a shared method to avoid duplicated code.

This should also be heavily commented to explain why it exists.